copy/progress_reader_test.go

```diff
@@ -0,0 +1,94 @@
+package copy
+
+import (
+	"bytes"
+	"testing"
+	"time"
+
+	"github.com/containers/image/v4/types"
+	"github.com/stretchr/testify/assert"
+)
+
+func newSUT(
+	t *testing.T,
+	duration time.Duration,
+	channel chan types.ProgressProperties,
+) *progressReader {
+	artifact := types.BlobInfo{Size: 10}
+
+	go func() {
```

Comment:
Note to self: does the testing framework know it has to wait for all spawned goroutines?